src/box/sql/build.c

```diff
@@ -2117,6 +2169,576 @@ sql_drop_table(struct Parse *parse_context)
+	}
+
+	struct ExprList *child_cols = parse->create_fk_def.child_cols;
+	char *value;
+	uint32_t size;
+	int id;
+	if (child_cols == NULL || child_cols->nExpr == 1) {
+		fk_to_format(parse, space->def);
+		id = BOX_SPACE_FIELD_FORMAT;
+		value = encode_format(&parse->region, space->def, &size);
+	} else {
+		fk_to_opts(parse, space->def);
+		id = BOX_SPACE_FIELD_OPTS;
+		value = encode_opts(&parse->region, space->def, &size);
+	}
+	if (value == NULL) {
+		parse->is_aborted = true;
+		return;
+	}
+	vdbe_emit_change_space_def(parse, space->def, value, size, id);
+}
+
+void
+sql_drop_field_constraint(struct Parse *parse)
+{
+	struct drop_constraint_def *drop_cdef = &parse->drop_constraint_def;
+	struct drop_entity_def *drop_def = &drop_cdef->base;
+	const char *space_name = drop_def->base.entity_name->a[0].zName;
+	assert(space_name != NULL);
+	struct space *space = space_by_name(space_name);
+	if (space == NULL) {
+		diag_set(ClientError, ER_NO_SUCH_SPACE, space_name);
+		parse->is_aborted = true;
+		return;
+	}
+
+	const char *field_name =
+		sql_normalized_name_region_new(&parse->region,
+					       drop_cdef->field_name.z,
+					       drop_cdef->field_name.n);
+	if (field_name == NULL) {
+		parse->is_aborted = true;
+		return;
+	}
+	const char *name =
+		sql_normalized_name_region_new(&parse->region, drop_def->name.z,
+					       drop_def->name.n);
+	if (name == NULL) {
+		parse->is_aborted = true;
+		return;
+	}
+
+	uint32_t fieldno;
+	struct space_def *def = space->def;
+	for (fieldno = 0; fieldno < def->field_count; ++fieldno) {
+		if (strcmp(field_name, def->fields[fieldno].name) == 0)
+			break;
+	}
+	const char *cname = tt_sprintf("%s.%s", field_name, name);
+	if (fieldno == def->field_count) {
+		diag_set(ClientError, ER_NO_SUCH_CONSTRAINT, cname, def->name);
+		parse->is_aborted = true;
+		return;
+	}
+	struct field_def *field = &def->fields[fieldno];
+
+	struct tuple_constraint_def *old_cdef = field->constraint_def;
+	uint32_t cid;
+	for (cid = 0; cid < field->constraint_count; ++cid) {
+		if (strcmp(old_cdef[cid].name, name) == 0)
+			break;
+	}
+	if (cid == field->constraint_count) {
+		diag_set(ClientError, ER_NO_SUCH_CONSTRAINT, cname, def->name);
+		parse->is_aborted = true;
+		return;
+	}
+
+	space = sql_shallow_space_copy(parse, space);
+	if (space == NULL) {
+		parse->is_aborted = true;
+		return;
+	}
+	def = space->def;
+	field = &def->fields[fieldno];
+	--field->constraint_count;
+	struct tuple_constraint_def *cdef = NULL;
+	if (field->constraint_count != 0) {
+		uint32_t count = field->constraint_count;
+		cdef = constraint_drop(parse, old_cdef, count, cid);
+		if (cdef == NULL)
+			return;
+		field->constraint_def = cdef;
+	}
+	int id = BOX_SPACE_FIELD_FORMAT;
+	uint32_t size;
+	char *value = encode_format(&parse->region, def, &size);
+	if (value == NULL) {
+		parse->is_aborted = true;
+		return;
+	}
+	vdbe_emit_change_space_def(parse, space->def, value, size, id);
+}
+
+bool
+is_tuple_constraint(struct Parse *parse)
```

Comment:
When a function is not static, usually it should have more strict name. For instance, here is at least clearly an sql function, so `sql_is_tuple_constraint()`. Then, it operates on parser, so `sql_parser_is_tuple_constraing()` or whatever prefix we use for parser "methods". Then it works only for dropping constraints, so it becomes `sql_parser_is_dropping_tuple_constraint()`. Having `is_` functions in the global namespace doesn't look safe, it will clash with something eventually for sure. The final name you might want to shorten somehow anyway. But the current one doesn't tell almost anything about what it takes, when can work.

Reply:
Understood. In the current version, I don't need this function anymore, but I'll keep this comment in mind, thanks.